CRM/Membershipextrasimporterapi/EntityImporter/Membership.php

```diff
@@ -37,9 +40,39 @@ public function import() {
 
     $membershipId = $this->getMembershipIdIfExist();
     if ($membershipId) {
-      return $membershipId;
+      $this->updateExistingMembership($membershipId);
+    }
+    else {
+      $membershipId = $this->createNewMembership();
+    }
+
+    return $membershipId;
+  }
+
+  private function getMembershipIdIfExist() {
+    $sqlQuery = "SELECT entity_id as id FROM civicrm_value_membership_ext_id WHERE external_id = %1";
+    $membershipId = SQLQueryRunner::executeQuery($sqlQuery, [1 => [$this->rowData['membership_external_id'], 'String']]);
```

Comment:
Is 'id' a String?  Shouldn't it be integer?

Reply:
it is an external id, which means something coming from an external system, and different systems not necessary define ids as integers, they could be strings